modules/client-common/src/main/java/org/apache/ignite/internal/client/proto/ClientComputeJobUnpacker.java

```diff
@@ -31,68 +31,83 @@
-        return unpack(marshaller, unpacker);
-    }
-
     /**
      * Unpacks compute job result. If the marshaller is provided, it will be used to unmarshal the result. If the marshaller is not provided
-     * and the result is a native column type or a tuple, it will be unpacked accordingly.
+     * and the result class is provided and the result is a tuple, it will be unpacked as a pojo of that class. If the marshaller is not
+     * provided and the result is a native column type or a tuple, it will be unpacked accordingly.
      *
-     * @param marshaller Marshaller.
      * @param unpacker Unpacker.
+     * @param marshaller Marshaller.
+     * @param resultClass Result class.
      * @return Unpacked result.
      */
-    public static @Nullable Object unpackJobResult(@Nullable Marshaller<?, byte[]> marshaller, ClientMessageUnpacker unpacker) {
-        return unpack(marshaller, unpacker);
-    }
-
-    /** Underlying byte array expected to be in the following format: | typeId | value |. */
-    private static @Nullable Object unpack(@Nullable Marshaller<?, byte[]> marshaller, ClientMessageUnpacker unpacker) {
+    public static @Nullable Object unpackJobResult(
+            ClientMessageUnpacker unpacker,
+            @Nullable Marshaller<?, byte[]> marshaller,
+            @Nullable Class<?> resultClass
+    ) {
         if (unpacker.tryUnpackNil()) {
             return null;
         }
 
+        // Underlying byte array expected to be in the following format: | typeId | value |.
         int typeId = unpacker.unpackInt();
 
         switch (typeId) {
-            case NATIVE_ID:
+            case NATIVE:
                 if (marshaller != null) {
                     throw new UnmarshallingException(
                             "Can not unpack object because the marshaller is provided but the object was packed without marshaller."
                     );
                 }
 
                 return unpacker.unpackObjectFromBinaryTuple();
-            case MARSHALLED_TUPLE_ID:
+            case MARSHALLED_TUPLE:
                 return TupleWithSchemaMarshalling.unmarshal(unpacker.readBinary());
 
-            case MARSHALLED_OBJECT_ID:
+            case MARSHALLED_CUSTOM:
                 if (marshaller == null) {
                     throw new UnmarshallingException(
                             "Can not unpack object because the marshaller is not provided but the object was packed with marshaller."
                     );
                 }
                 return tryUnmarshalOrCast(marshaller, unpacker.readBinary());
 
-            case MARSHALLED_POJO_ID:
-                return unpackPojo(unpacker);
+            case MARSHALLED_POJO:
+                if (resultClass == null) {
+                    throw new UnmarshallingException(
+                            "Can not unpack object because the pojo class is not provided but the object was packed as pojo."
```

Comment:
```suggestion
                            "Can not unpack object because the pojo class is not provided but the object was packed as pojo. Provide Job result type in JobDescriptor.resultClass."
```